v2/event/event_unmarshal.go

```diff
@@ -385,9 +386,13 @@ func consumeData(e *Event, isBase64 bool, iter *jsoniter.Iterator) error {
 		e.DataBase64 = true
 
 		// Allocate payload byte buffer
-		base64Encoded := iter.ReadStringAsSlice()
+		base64Encoded := iter.ReadString()
+		err := json.Unmarshal([]byte(`"`+base64Encoded+`"`), &base64Encoded)
```

Comment:
could there be issues here if `base64Encoded` already contains surrounding `""` ? For example, if it's a JSON-encoded string e.g., `"{\"hello\":\"world\"}"`?

Reply:
nit: I don't prefer the in-place mutation of the `base64Encoded` variable, better use dedicated variables in case the code is later extended and another developer relies on the original `base64Encoded` without noticing the overwrite.

Reply:
re: extra quotes... I think we're ok because any " in the string would need to be escaped, as you've done it. The quotes around your json object/string thingy are removed by the iter.ReadString() call, so just `{\"hello\":\"world\"}` is passed into Unmarshal (after being re-wrapped with quotes).

I added another variable for the Unmarshal per your suggestion.

Reply:
Also, this is only for base64 encoded stuff, not normal json strings.